### DISCUSSION TITLE
Try to mine a block in the zcash testnet

### DISCUSSION THREAD

**Discussion by oxarbitrage** [ZcashFoundation/zebra#5804] [Ideas]
Here we want to discuss and propose ideas to create shorter term goal for the getblocktemplate work in order to somehow "mine" a block in the testnet.

What gets proposed here should be added to https://github.com/ZcashFoundation/zebra/issues/5803

**Comment by teor2345** [comment]
I think we can do testing using a block proposal, which is a block without a proof of work.

Here's how we could do that:
1. Finish all the required getblocktemplate RPC fields
2. Create a block header from the RPC fields, with a zeroed-out equihash solution, and serialize it
3. Create the block transactions by combining all the coinbasetxn and transaction `data` fields
4. Send that block to `zcashd` using `getblocktemplate '{ "mode": "proposal", "data": "(hex data)" }'

`zcashd` has proposal support, but it's not documented. Here's the spec:
https://en.bitcoin.it/wiki/BIP_0023#Block_Proposal

**Comment by oxarbitrage** [comment]
Thanks @teor2345 that looks good. 

@arya2 do you have any additional comments to what Teor is proposing ? I want to add that to the details section of https://github.com/ZcashFoundation/zebra/issues/5803

**Comment by arya2** [comment]
Nope, that plan looks good.

**Comment by mpguerra** [comment]
Any other potential ideas to discuss? Should be consider this discussion closed?
